### ISSUE TITLE
build errors on macOS 14.2

### ISSUE THREAD

**Issue by artkiver** [OpenSMTPD/OpenSMTPD#1232]
Hi there,

I am not the MacPorts OpenSMTPD maintainer, yet I thought I would give a try at updating the MacPorts' Portfile for OpenSMTPD's 7.4.0p0 release.

While previous minor modifications went relatively smoothly, I have run into errors using that methodology and even attempting to build from the source in the tarball using the instructions from the README.md.

I've shared the error logs here:
https://trac.macports.org/ticket/68573

For the sake of completeness, I thought that it might be worth mentioning upstream as well? My apologies in advance if this is related to a local build issue (though I did at least test to make sure that this system can build 7.3.0p2 successfully).

**Comment by ryandesign** [comment]
Looks like opensmtpd is trying to declare a function `strlcpy` via its bundled copy of openbsd-compat, despite the fact that macOS already has `strlcpy`, which the configure script correctly identified.

**Comment by omar-polo** [comment]
Hello,

It was caused by ebea68918a994ddaa43bb1aac769bc086f52e2f2, I forgot to update the `#ifdef` from `HAVE_DECL_STRLCPY` to `HAVE_STRLCPY` (and similarly for strlcat).  Now it should be fixed.

Thanks for the report!

**Comment by omar-polo** [comment]
Can I consider this to be fixed?  We'll release a p1 in a few days, I'd like to have it working for macos too.

Thanks!

**Comment by artkiver** [comment]
> Can I consider this to be fixed? We'll release a p1 in a few days, I'd like to have it working for macos too.
> 
> Thanks!

Shoot, my apologies for not testing this sooner!

I seem to be running into a different issue now? (assuming I copied the updated openbsd-compat.h locally correctly? I'm not quite a git/GitHub wizard) log attached.
[opensmtpd7.4.0p0.5.log](https://github.com/OpenSMTPD/OpenSMTPD/files/13246001/opensmtpd7.4.0p0.5.log)

**Comment by omar-polo** [comment]
@artkiver 
> Shoot, my apologies for not testing this sooner!

No problem, it's my fault for hurrying :)

> I seem to be running into a different issue now?

Yes, there's some issues with how the discovery for b64_ntop is handled.  Will look into it soon.  (I should definitely add a macos machine in the CI!)

**Comment by omar-polo** [comment]
7.4.0p1 is out now and includes also these fixes, so I'm going to close this issue.  Of course, feel free to reopen or file a new one if other problems arise.

Thanks!

**Comment by artkiver** [comment]
> 7.4.0p1 is out now and includes also these fixes, so I'm going to close this issue. Of course, feel free to reopen or file a new one if other problems arise.
> 
> Thanks!

Awesome! Thank you!

I confirm testing locally that it 7.4.0p1 is building correctly (though now I'm testing on 14.2 beta 3).

I needed to remove one line from the MacPorts Portfile to make it happy, but that is documented here:
https://trac.macports.org/ticket/68573#comment:4

I'm still not the MacPorts maintainer for OpenSMTPD though so I will probably hold off on submitting a PR for a little bit to see if the maintainer has some other preference about the manpage change I made. Regardless, that isn't super salient for you as an upstream maintainer. Thank you again!

**Comment by omar-polo** [comment]
@artkiver the `--with-mantype` failure is... my fault.  I'm a bit unlucky with the releases recently as you can see.

There's a typo in the configure.ac https://github.com/OpenSMTPD/OpenSMTPD/commit/6c9d150be58df090400a22ed30dcf275cc3458ad that breaks the option handling.

I don't know much about macos and macports in general.  Is carring a patch for some months a big deal?  There are problems with mandoc manpages on mac?  The next release will have the flag fixed again, just wondering whether this error warrants another release or not.

Thank you!

**Comment by artkiver** [comment]
Ah, at least it was a simple fix? I submitted a PR to update the MacPort here: https://github.com/macports/macports-ports/pull/21431

Albeit, slightly different since it is patching configure and configure.ac?

Hopefully that is acceptable! I know that in some other MacPorts, I've downstreamed patches a little bit ahead of an officially tagged release (e.g. a workaround for zlib 1.3 and OpenSSH before 9.5 was released, since MacPorts had updated its zlib) and then removed such things as upstream versions have been tagged. Though, as I mentioned previously, I am not the OpenSMTPD MacPorts maintainer, nor do I have commit access to that project, so it's up to others to determine whether my PR is worth merging.

Thank you for going the extra mile to look at the MacPorts Trac and ascertain the typo and rectify it anyway! I'm still relatively a novice when it comes to MacPorts myself relative to many of its other contributors.

TL;DR: it probably doesn'T require another OpenSMTPD release for the time being? At least from the MacPorts angle, my local patches should suffice, and presumably I or the maintainer or someone else will remove that workaround when another OpenSMTPD release is tagged with the included fix obviating the need for such patches downstream.
